// translate-google-modified/languages.js
/*
* @Author: shikar
* @Date:   2017-02-05 15:29:19
* @Last Modified by:   shikar
* @Last Modified time: 2017-02-06 13:40:25
*/
'use strict'
const langs = {
  auto: ['automatic', 'automatique'],
  af: ['afrikaans'],
  sq: ['albanian', 'albanais'],
  ar: ['arabic', 'arabe', 'bomb'],
  hy: ['armenian', 'arménien'],
  az: ['azerbaijani'],
  eu: ['basque'],
  be: ['belarusian', 'biélorusse', 'bielorusse'],
  bn: ['bengali'],
  bs: ['bosnian'],
  bg: ['bulgarian', 'bulgare'],
  ca: ['catalan'],
  ceb: ['cebuano'],
  ny: ['chichewa'],
  'zh-cn': ['chinese simplified', 'chinois', 'ch'],
  'zh-tw': ['chinese traditional'],
  co: ['corsican'],
  hr: ['croatian'],
  cs: ['czech'],
  da: ['danish', 'danois'],
  nl: ['dutch', 'néerlandais', 'neerlandais', 'hollandais'],
  en: ['english', 'anglais', 'us', 'usa'],
  eo: ['esperanto'],
  et: ['estonian'],
  tl: ['filipino'],
  fi: ['finnish'],
  fr: ['french', 'francais', 'français'],
  fy: ['frisian'],
  gl: ['galician'],
  ka: ['georgian'],
  de: ['german', 'allemand', 'al'],
  el: ['greek'],
  gu: ['gujarati'],
  ht: ['haitian creole'],
  ha: ['hausa'],
  haw: ['hawaiian'],
  iw: ['hebrew'],
  hi: ['hindi'],
  hmn: ['hmong'],
  hu: ['hungarian'],
  is: ['icelandic'],
  ig: ['igbo'],
  id: ['indonesian'],
  ga: ['irish'],
  it: ['italian', 'italien'],
  ja: ['japanese', 'japonais', 'jap'],
  jw: ['javanese'],
  kn: ['kannada'],
  kk: ['kazakh'],
  km: ['khmer'],
  ko: ['korean', 'coréen', 'coreen'],
  ku: ['kurdish (kurmanji)'],
  ky: ['kyrgyz'],
  lo: ['lao'],
  la: ['latin'],
  lv: ['latvian'],
  lt: ['lithuanian', 'lituanien'],
  lb: ['luxembourgish'],
  mk: ['macedonian'],
  mg: ['malagasy', 'malgache', 'gay'],
  ms: ['malay'],
  ml: ['malayalam'],
  mt: ['maltese'],
  mi: ['maori'],
  mr: ['marathi'],
  mn: ['mongolian', 'mongol', 'mongole'],
  my: ['myanmar (burmese)'],
  ne: ['nepali'],
  no: ['norwegian'],
  ps: ['pashto'],
  fa: ['persian'],
  pl: ['polish', 'polonais'],
  pt: ['portuguese', 'portugais', 'portugay'],
  ma: ['punjabi'],
  ro: ['romanian', 'roumain'],
  ru: ['russian', 'russe', 'vodka'],
  sm: ['samoan'],
  gd: ['scots gaelic'],
  sr: ['serbian'],
  st: ['sesotho'],
  sn: ['shona'],
  sd: ['sindhi'],
  si: ['sinhala'],
  sk: ['slovak'],
  sl: ['slovenian'],
  so: ['somali'],
  es: ['spanish', 'espagnol', 'espanol', 'español'],
  su: ['sudanese'],
  sw: ['swahili'],
  sv: ['swedish', 'suédois'],
  tg: ['tajik'],
  ta: ['tamil'],
  te: ['telugu'],
  th: ['thai'],
  tr: ['turkish', 'turc'],
  uk: ['ukrainian', 'ukrainien'],
  ur: ['urdu'],
  uz: ['uzbek'],
  vi: ['vietnamese', 'vietnamien'],
  cy: ['welsh'],
  xh: ['xhosa'],
  yi: ['yiddish'],
  yo: ['yoruba'],
  zu: ['zulu', 'zoulou']
}

function getCode(desiredLang) {
  if (!desiredLang) {
    return false
  }
  desiredLang = desiredLang.toLowerCase()
  if (langs[desiredLang]) {
    return desiredLang
  }
  var keys = Object.keys(langs).filter(key => Array.isArray(langs[key]) && langs[key].includes(desiredLang));
  return keys[0] || false
}

function isSupported(desiredLang) {
  return Boolean(getCode(desiredLang))
}

module.exports = langs
module.exports.isSupported = isSupported
module.exports.getCode = getCode
